geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java

```diff
@@ -300,14 +299,18 @@ public RedisKeyCommands getKeyCommands() {
     return keyCommands;
   }
 
+  @SuppressWarnings("unchecked")
   public Set<DistributedMember> getRegionMembers() {
-    PartitionRegionInfo info = PartitionRegionHelper.getPartitionRegionInfo(dataRegion);
-    Set<DistributedMember> membersWithDataRegion = new HashSet<>();
-    for (PartitionMemberInfo memberInfo : info.getPartitionMemberInfo()) {
-      membersWithDataRegion.add(memberInfo.getDistributedMember());
+    InternalDistributedMember myId =
+        partitionedRegion.getDistributionManager().getDistributionManagerId();
+    Set<InternalDistributedMember> otherIds =
+        partitionedRegion.getRegionAdvisor().adviseDataStore();
+    if (otherIds.isEmpty()) {
+      return singleton(myId);
+    } else {
+      otherIds.add(myId);
+      return (Set<DistributedMember>) (Set<?>) otherIds;
```

Comment:
Not very important, but I think you can call `adviseDataStore(true)` here and then avoid this empty check. It's the same thing essentially but would avoid folks wondering why it's done this way.

Reply:
I reworked this so that this method only returns remote members and the caller of it executes the publish directly for the local member and only using the FunctionService for remote members.